controller.go

```diff
@@ -0,0 +1,55 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package weaver
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ServiceWeaver/weaver/runtime/protos"
+)
+
+// controller is a component hosted in every weavelet. Deployers make calls to this component to
+// fetch information about the weavelet, and to make it do various things.
+//
+// Arguments and results are protobufs to allow deployers to evolve independently of application
+// binaries.
+type controller interface {
```

Comment:
Right now, the `controller` interface is duplicated here and in `envelope.go`. We could do the following to remove some of the duplication. 

First, create an `internal/control` package with a `controller.go` file:

```go
package control

import (
        "context"

        "github.com/ServiceWeaver/weaver/runtime/protos"
)

type Controller interface {
        GetHealth(context.Context, *protos.GetHealthRequest) (*protos.GetHealthReply, error)
        UpdateComponents(context.Context, *protos.UpdateComponentsRequest) (*protos.UpdateComponentsReply, error)
}
```

Second, update the `controller` definition here to the following:

```go
type controller control.Controller
```

Third, delete the `controller` interface in `envelope.go`. Additionally, replace all references to `controller` in `envelope.go` with `control.Controller`. 

Fourth, add the following line to `remoteweavelet.go`:

```go
var _ control.Controller = (*RemoteWeavelet)(nil)
```

What do you think? It's all a bit convoluted, but it does prevent us from changing the `controller` interface and forgetting to update either `RemoteWeavelet` or `Envelope`.

Reply:
Great idea! Done.